cognite/client/data_classes/workflows.py

```diff
@@ -1372,6 +1373,15 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
     @classmethod
     def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> Self:
         trigger_type = resource["triggerType"]
+        # Special handling for schedule trigger with timezone
+        if trigger_type == "schedule" and "timezone" in resource and resource["timezone"]:
+            resource = dict(resource)  # avoid mutating input
+            timezone_value = resource["timezone"]
+            # Convert timezone to ZoneInfo if it's a string, leave as-is if already ZoneInfo
+            if isinstance(timezone_value, str):
+                resource["timezone"] = ZoneInfo(timezone_value)
```

Comment:
Why is this logic not in the `.load_trigger` method in the `WorkflowScheduledTriggerRule`. It is strange that it is in the base class for TriggerRules.